miden-lib/src/transaction/memory.rs

```diff
@@ -253,20 +257,20 @@ pub const NATIVE_ACCT_STORAGE_SLOTS_SECTION_PTR: MemoryAddress =
 // ================================================================================================
 
 /// The size of the memory segment allocated to each note.
-pub const NOTE_MEM_SIZE: MemoryAddress = 512;
+pub const NOTE_MEM_SIZE: MemoryAddress = 2048;
 
 #[rustfmt::skip]
 // INPUT NOTES DATA
 // ------------------------------------------------------------------------------------------------
 // Inputs note section contains data of all notes consumed by a transaction. The section starts at
-// memory offset 1_048_576 with a word containing the total number of input notes and is followed
+// memory offset 4_194_304 with a word containing the total number of input notes and is followed
 // by note nullifiers and note data like so:
 //
-// ┌─────────┬───────────┬───────────┬─────┬───────────┬─────────┬────────┬────────┬─────┬────────┐
-// │   NUM   │  NOTE 0   │  NOTE 1   │ ... │  NOTE n   │ PADDING │ NOTE 0 │ NOTE 1 │ ... │ NOTE n │
-// │  NOTES  │ NULLIFIER │ NULLIFIER │     │ NULLIFIER │         │  DATA  │  DATA  │     │  DATA  │
-// └─────────┴───────────┴───────────┴─────┴───────────┴─────────┴────────┴────────┴─────┴────────┘
-//  1_048_576  1_048_577   1_048_578        1_048_576+n      1_064_960   +512    +1024  +512n
+// ┌─────────┬───────────┬───────────┬─────┬───────────┬─────────┬────────┬────────┬───────┬────────┐
+// │   NUM   │  NOTE 0   │  NOTE 1   │ ... │  NOTE n   │ PADDING │ NOTE 0 │ NOTE 1 │  ...  │ NOTE n │
+// │  NOTES  │ NULLIFIER │ NULLIFIER │     │ NULLIFIER │         │  DATA  │  DATA  │       │  DATA  │
+// └─────────┴───────────┴───────────┴─────┴───────────┴─────────┴────────┴────────┴───────┴────────┘
+//  4_194_304  4_194_308   4_194_312        4_194_304+4n     4_259_840  +2048    +4096   +2048n
```

Comment:
```suggestion
//  4_194_304  4_194_308   4_194_312        4_194_304+4(n+1)     4_259_840  +2048    +4096   +2048(n+1)
```

Nit: For accuracy, this would have to be `n+1`.

Reply:
Indeed!